File: my-folder/0074-search-a-2d-matrix/solution.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        rows = len(matrix)
        cols = len(matrix[0])
        row, col = 0, cols - 1

        
        while row < rows and col > -1:
            curr = matrix[row][col]
            if curr == target:
                return True
            if curr < target:
                row += 1
            else:
                col -= 1
        return False
                


            

File: my-folder/0074-search-a-2d-matrix/solution.c
#include <stdbool.h>

bool searchMatrix(int** matrix, int matrixSize, int* matrixColSize, int target) {
    int rows = matrixSize;
    int cols = matrixColSize[0];
    int row = 0;
    int col = cols - 1;

    while (row < rows && col >= 0) {
        int curr = matrix[row][col];
        if (curr == target) {
            return true;
        } else if (curr < target) {
            row++;
        } else {
            col--;
        }
    }

    return false;
}

